### PR TITLE
prod rework + docker networks

### LINKED ISSUE #530: feat: Implement docker networks
**Is your feature request related to a problem? Please describe.**
Currently we only use one default network for all communications

**Describe the solution you'd like**
Create separate networks for different use cases, such as communicating with DB and load-balancers. Will enable moving to a global traefik-based load balancer at a later point

**Additional context**
https://www.warp.dev/terminus/docker-compose-networks

### PR DESCRIPTION
**Pull request by Theoabw**

### CODE REVIEW

**Comment by theoabw** [comment]
Continues from #524

**Comment by jaspnas** [comment]
Encountering connection or DNS problems when testing with WSL, May be Windows being Windows or a race condition during startup. 

Basically nginx can't always find upstream

File: docker-compose.prod.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    volumes:
+      - date_postgres_data:/var/lib/postgresql/data/
+    ports:
+      - ${DATE_DB_PORT}:5432
+    environment:
+      - POSTGRES_PASSWORD=${DATE_DB_PASSWORD}
+    networks:
+      - internal
+  web:
+    build: .
```
Comment (jaspnas) [inline docker-compose.prod.yml]: Should not have build in prod

**Review by jaspnas — CHANGES_REQUESTED** [review]

**Comment by jaspnas** [comment]
Seems like DNS problems is just docker desktop on windows being unstable, going to test on my linux laptop tomorrow

**Review by jaspnas — APPROVED** [review]
Seems to work reliably on linux

image

**Pull request merged**